src/ui/public/vislib/lib/dispatch.js

```diff
@@ -101,9 +101,7 @@ export default function DispatchClass(Private, config) {
       const isClickable = this.listenerCount('click') > 0;
       const addEvent = this.addEvent;
       const $el = this.handler.el;
-      if (!this.handler.highlight) {
-        this.handler.highlight = self.highlight;
-      }
+      this.handler.highlight = self.highlight;
```

Comment:
so you are always overwriting handler.highlight, even if its already defined. when [area chart defines highlight function on handler](https://github.com/elastic/kibana/blob/master/src/ui/public/vislib/visualizations/area_chart.js#L34) it will be ignored ....